kafka-rest/src/main/java/io/confluent/kafkarest/resources/ConsumerGroupsResource.java

```diff
@@ -0,0 +1,217 @@
+   * @return List of group names with group coordinator host of each group.
+   *     [{"groupId":"testGroup", "coordinator": {"host": "127.0.0.1", "port": "123"}}]
+   */
+  @GET
+  @PerformanceMetric("groups.list")
+  public List<ConsumerGroup> list(@QueryParam("page_offset") Integer pageOffset,
+                                  @QueryParam("page_size") Integer pageSize)
+          throws Exception {
+    final boolean needPartOfData = Optional.ofNullable(pageOffset).isPresent()
+        && Optional.ofNullable(pageSize).isPresent()
+        && pageSize > 0;
+    if (needPartOfData) {
+      return context.getGroupMetadataObserver()
+          .getPagedConsumerGroupList(pageOffset, pageSize);
+    }
+    return context.getGroupMetadataObserver().getConsumerGroupList();
+  }
+
+  /**
+   * <p>Get partitions list for group groupId</p>
+   * <p>Example: http://127.0.0.1:2081/groups/testGroup/partitions</p>
+   *
+   * @param groupId - Group name.
+   * @return Consumer subscription information. Include group offsets, lags and
+   *      group coordinator information.
+   *      { "topicPartitionList":[
+   *         { "consumerId":"consumer-1-88792db6-99a2-4064-aad2-38be12b32e88",
+   *            "consumerIp":"/{some_ip}",
+   *            "topicName":"1",
+   *            "partitionId":0,
+   *            "currentOffset":15338734,
+   *            "lag":113812,
+   *            "endOffset":15452546},
+   *         { "consumerId":"consumer-1-88792db6-99a2-4064-aad2-38be12b32e88",
+   *            "consumerIp":"/{some_ip}",
+   *            "topicName":"1",
+   *            "partitionId":1,
+   *            "currentOffset":15753823,
+   *            "lag":136160,
+   *            "endOffset":15889983},
+   *         { "consumerId":"consumer-1-88792db6-99a2-4064-aad2-38be12b32e88",
+   *            "consumerIp":"/{some_ip}",
+   *            "topicName":"1",
+   *            "partitionId":2,
+   *            "currentOffset":15649419,
+   *            "lag":133052,
+   *            "endOffset":15782471}],
+   *        "topicPartitionCount":3,
+   *        "coordinator":{ "host":"{coordinator_host_name}","port":9496 }
+   *      }
+   */
+  @GET
+  @Path("/{groupId}/partitions")
+  @PerformanceMetric("groups.get.partitions")
+  public ConsumerGroupSubscription getPartitionsInformation(@PathParam("groupId") String groupId)
+          throws Exception {
+    return context.getGroupMetadataObserver().getConsumerGroupInformation(groupId);
+  }
+
+  /**
+   * <p>Get topics list for group groupId</p>
+   * <p>Example: http://127.0.0.1:2081/groups/testGroup/topics</p>
+   *
+   * @param groupId - Group name.
+   * @param pageSize - Optional parameter. Use for paging.
+   *             Restrict count of returned entities with group information.
+   * @param pageOffset - Optional parameter. Use for paging.
+   *               Offset which starts return records from.
+   * @return Topic names who read by specified consumer group.
+   *       [{"name":"1"}]
```

Comment:
Does Jackson omit empty values?

Reply:
Jackson does not omit null-values by default. It omit these values if we add annotation to class Topic
`@JsonInclude(JsonInclude.Include.NON_EMPTY)`
or
`@JsonInclude(JsonInclude.Include.NON_NULL)`